pkg/controller/job/job_controller_test.go

```diff
@@ -1048,6 +1049,25 @@ func TestControllerSyncJob(t *testing.T) {
 			podIndexLabelDisabled:  true,
 			expectedReady:          ptr.To[int32](0),
 		},
+		"FailureTarget=False condition added manually": {
+			parallelism: 1,
+			completions: 1,
+			activePods:  1,
+			readyPods:   1,
+			initialStatus: &jobInitialStatus{
+				active: 1,
+				startTime: func() *time.Time {
+					now := time.Now()
+					return &now
+				}(),
+				conditions: []batch.JobCondition{
+					*newCondition(batch.JobFailureTarget, v1.ConditionFalse, "ConditionAddedManually", "Testing", realClock.Now()),
+				},
+			},
+			expectedActive:      1,
+			expectedReady:       ptr.To[int32](1),
+			jobPodFailurePolicy: true,
```

Comment:
Maybe move enabling of the feature to the top so that it is a precondition. At the bottom we place asserts typically.

Reply:
Fine for me! I just followed the order in the [test struct](https://github.com/kubernetes/kubernetes/blob/0c8b3e5f305bf2bf56d47019199b81330d90c2c3/pkg/controller/job/job_controller_test.go#L292)